PRESUBMIT.py

```diff
@@ -280,6 +280,48 @@ def CheckLicense(input_api, output_api):
     return result
 
 
+def CheckNewSourceFileWithoutGnChangeOnUpload(input_api, output_api):
+    """Checks newly added .cc/.h/.mm files have corresponding GN changes."""
+    files_to_check = (
+        r'.+\.cc$',
+        r'.+\.h$',
+        r'.+\.mm$',
+    )
+
+    def new_sources(f):
+        return input_api.FilterSourceFile(f, files_to_check=files_to_check)
+
+    new_sources = []
+    for f in input_api.AffectedSourceFiles(new_sources):
+        if f.Action() != 'A':
+            continue
+        new_sources.append(f.LocalPath())
+
+    def gn_files(f):
+        return input_api.FilterSourceFile(f, files_to_check=(r'.+\.gn', ))
```

Comment:
can you also test that this covers `.gn` and `.gni` files.

Reply:
I think it was but might involve other stuff by accident, changed to .+\.gn$ and .+\.gni$